samples/subsys/usb/hid-cdc/src/main.c

```diff
@@ -0,0 +1,868 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <device.h>
+#include <gpio.h>
+#include <uart.h>
+#include <string.h>
+#include <random/rand32.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+#include <usb/class/usb_cdc.h>
+
+#define LOG_LEVEL LOG_LEVEL_DBG
+LOG_MODULE_REGISTER(main);
+
+#ifdef SW0_GPIO_CONTROLLER
+#define PORT0 SW0_GPIO_CONTROLLER
+#else
+#error SW0_GPIO_CONTROLLER needs to be set
+#endif
+
+#ifdef SW0_GPIO_PIN
+#define PIN0     SW0_GPIO_PIN
+#else
+#error SW0_GPIO_PIN needs to be set
+#endif
+
+#ifdef SW0_GPIO_FLAGS
+#define PIN0_FLAGS SW0_GPIO_FLAGS
+#else
+#error SW0_GPIO_FLAGS needs to be set
+#endif
+
+#ifdef SW1_GPIO_PIN
+#define PIN1	SW1_GPIO_PIN
+#endif
+
+#ifdef SW1_GPIO_CONTROLLER
+#define PORT1	SW1_GPIO_CONTROLLER
+#endif
+
+#ifdef SW1_GPIO_FLAGS
+#define PIN1_FLAGS SW1_GPIO_FLAGS
+#endif
+
+#ifdef SW2_GPIO_PIN
+#define PIN2	SW2_GPIO_PIN
+#endif
+
+#ifdef SW2_GPIO_CONTROLLER
+#define PORT2	SW2_GPIO_CONTROLLER
+#endif
+
+#ifdef SW2_GPIO_FLAGS
+#define PIN2_FLAGS SW2_GPIO_FLAGS
+#endif
+
+#ifdef SW3_GPIO_PIN
+#define PIN3	SW3_GPIO_PIN
+#endif
+
+#ifdef SW3_GPIO_CONTROLLER
+#define PORT3	SW3_GPIO_CONTROLLER
+#endif
+
+#ifdef SW3_GPIO_FLAGS
+#define PIN3_FLAGS SW3_GPIO_FLAGS
+#endif
+
+#define LED_PORT	LED0_GPIO_CONTROLLER
+#define LED		LED0_GPIO_PIN
+
+/* Event FIFO */
+
+K_FIFO_DEFINE(evt_fifo);
+
+enum evt_t {
+	GPIO_BUTTON_0	= 0x00,
+	GPIO_BUTTON_1	= 0x01,
+	GPIO_BUTTON_2	= 0x02,
+	GPIO_BUTTON_3	= 0x03,
+	CDC_UP		= 0x04,
+	CDC_DOWN	= 0x05,
+	CDC_LEFT	= 0x06,
+	CDC_RIGHT	= 0x07,
+	CDC_UNKNOWN	= 0x08,
+	CDC_STRING	= 0x09,
+	HID_MOUSE_CLEAR	= 0x0A,
+	HID_KBD_CLEAR	= 0x0B,
+	HID_KBD_STRING	= 0x0C,
+};
+
+struct app_evt_t {
+	sys_snode_t node;
+	struct k_mem_block block;
+	enum evt_t event_type;
+};
+
+#define FIFO_ELEM_MIN_SZ        sizeof(struct app_evt_t)
+#define FIFO_ELEM_MAX_SZ        sizeof(struct app_evt_t)
+#define FIFO_ELEM_COUNT         255
+#define FIFO_ELEM_ALIGN         sizeof(unsigned int)
+
+K_MEM_POOL_DEFINE(event_elem_pool, FIFO_ELEM_MIN_SZ, FIFO_ELEM_MAX_SZ,
+		  FIFO_ELEM_COUNT, FIFO_ELEM_ALIGN);
+
+
+static inline void app_evt_free(struct app_evt_t *ev)
+{
+	k_mem_pool_free(&ev->block);
+}
+
+static inline void app_evt_put(struct app_evt_t *ev)
+{
+	k_fifo_put(&evt_fifo, ev);
+}
+
+static inline struct app_evt_t *app_evt_get(void)
```

Comment:
I am not sure what is the benefit of wrapping well known, one line `k_fifo_*` functions that everybody is familiar with, with these `app_evt_*` functions.